MD/md-3/asychronous-upgrades-problem.md

```diff
@@ -0,0 +1,81 @@
+# MCR: Asynchronous Upgrades, Fork Creation, and Fork Stake Problems
+
+The _Asynchronous Upgrades Problem_ is an issue with the [current implementation of MCR](https://github.com/movementlabsxyz/movement/tree/baa83356a14d44fd4e8346e1eddfc184cebc17d3/protocol-units/settlement/mcr) that arises when a partition of the network upgrades to a new version of the protocol while another partition remains on the old version. This can lead to failure to agree upon the current block height and currently would require the deployment of a new settlement contract to resolve.
+
+This state of MCR can arise in other contexts than upgrades. The use of the upgrade scenario is merely illustrative.
+
+It is resolvable by allowing network forks in a partial synchronous manner. However, this leads to a series of questions of when rewards and penalties should be applied to commitments on these forks. These questions are the crux of the [Fork Creation Problem](#fork-creation-problem) and [Fork Stake Problem](#fork-stake-problem).
+
+The conclusion drawn from the Fork Creation Problem is that penalties for creating a new fork cannot be implemented synchronously with the block commitments constituting that fork as well.
+
+The Fork Stake Problem suggests that rewards also cannot be implemented synchronously with their commitments, in order to ensure temporary forks do not have stake-based incentives.
+
+## Asynchronous Upgrades Problem
+
+Consider two honest partitions of the network A and B. At a given block height $h$, Partition A upgrades to a new version of the chain containing a state derivation $S_A(h)$. At the same block height, Partition B remains on the old version of the chain containing a state derivation $S_B(h)$ s.t. $S_A(h) \neq S_B(h)$.
+
+Because a fixed proportion of the stake is needed to agree on the current block height $h$, the network will not be able to agree on $h$.
+
+The network may be able to provisionally agree on $h\' > h$ within the block tolerance $\tau$ of the network. However, these agreements are never realized because $h$ remains undecided.
+
+## Solving the Problem with Partially Synchronous Forks
+
+The Asynchronous Upgrades Problem can be resolved by allowing the network to fork in a partially synchronous manner. That is, the network can agree at a defined point in time $t_{h - 1} + \epsilon$ to fork into two chains w/o loss of generality, one with state derivation $S_A(h)$ and the other with state derivation $S_B(h)$.
```

Comment:
@[apenzk](https://github.com/apenzk) apenzk [on Oct 10, 2024]

> I dont understand why we would introduce time here. why not use blockheight and run two instances of protocols P0 (old) and P1 (new). start signing with P1 at height H, continue (surviving hardfork) or discontinue (dead hardfork) with P0

@[l-monninger](https://github.com/l-monninger) l-monninger [on Oct 14, 2024]
> The situation above is one where an unforked set of commitments are being gathered at a given block height. To prevent the process of agreeing from lasting indefinitely, the fork choice would be made in time.

That is, block height is fixed, some other value needs to be considered for the decision to fork.